src/relay/op/tensor/transform.cc

```diff
@@ -51,6 +51,76 @@ namespace tvm {
 namespace relay {
 using tir::IntImmNode;
 
+TVM_REGISTER_NODE_TYPE(WindowsAttrs);
+
+bool WindowsRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
+                const TypeReporter& reporter) {
+  // `types` contains: [data, result]
+  ICHECK_EQ(types.size(), 2);
+  const auto* data = types[0].as<TensorTypeNode>();
+  if (data == nullptr) {
+    ICHECK(types[0].as<IncompleteTypeNode>())
+        << "windows: expect input type to be TensorType but get " << types[0];
```

Comment:
It would be better to use the diagnostic context to report this error. You can find an example here: https://github.com/apache/tvm/blob/main/src/relay/op/nn/nn.cc#L65-L68.

Reply:
Thanks, done!